flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerTest.java

```diff
@@ -415,13 +553,14 @@ public void testDeleteApplicationFiles() throws Exception {
 
 			resourceManager.deregisterApplication(ApplicationStatus.SUCCEEDED, null);
 			assertFalse("YARN application directory was not removed", Files.exists(applicationDir.toPath()));
+
+			stopResourceManager();
```

Comment:
Maybe we could add a function 
```
runTest(RunnableWithException testMethod) {
  startResourceManager();
  try {
    testMethod.run()
  } finally {
    stopResourceManager
  }
}
```

This method could then also be used by the `testOnContainerCompleted` test.